### ISSUE TITLE
Feature: Add scrollDebounceTime

### ISSUE THREAD

**Issue by AntonTrofimovich** [rintoj/ngx-virtual-scroller#244]
Hello. First of all thanks a lot for the lib. It's awesome :)

Would you like to add scrollDebounceTime functionality? It is similar to scrollThrottlingTime but starts timeout each time scroll event is fired.

Will be look like this:

```
debounceTrailing(func: Function, wait: number): Function {
	let timeout = undefined;
	const result = function () {
		const _this = this;
		const _arguments = arguments;

		if (timeout) {
			clearTimeout(timeout);
		}

		if (wait <= 0) {
			func.apply(_this, _arguments);
		} else {
			timeout = setTimeout(function () {
				timeout = undefined;
				func.apply(_this, _arguments);
			}, wait);
		}
	};

	return result;
}
```

Thanks in advance.

**Comment by speige** [comment]
I'm nervous that scrollDebounceTime will cause a confusing user experience. I honestly don't like scrollThrottlingTime either (it defaults to 0), but I think it's necessary in some scenarios for performance reasons.

The difference between throttling and debounce is that with throttling the UI will be forced to be updated on the interval, whereas with debounce the UI will remain stale as long as the user keeps scrolling. I feel like the user will become confused and think the UI is broken.

If you disagree, could you help me understand the value of adding debounce? What use case does it solve that's an issue with the current component?

**Comment by speige** [comment]
Thanks for your explanation. You're right, this could have value in some scenarios. I'll leave it up to the consuming application to make it obvious to the user why the screen is blank & that it'll load when they stop scrolling.

I've coded this & published to npm version 1.0.4. Please re-open if you continue to have issues.

**Comment by insomnia1337** [comment]
its nice feature but for users it might be a little confused that UI is broken. What do u think, when u have about 1000 records and want to get as much best performance?
